### DISCUSSION TITLE
Lesson 8: Always get rpc-error, json-rpc error (scenario explained)

### DISCUSSION THREAD

**Discussion by medhananda** [Cyfrin/foundry-full-course-cu#1051] [Q&A]
which step did I go wrong?

1. started anvil from command line 
2. imported an account let us call it account8 in metamsk
3. deployed the contract from forge/make file make deploy
4. Can see the contract address of the contract from anvil running log
5. The address from anvil log matches with the contract address displayed from make deploy
6. Now deployed index.html from the liveserver
7. clicked connect, I see account8 in metamsk
8. clicked on fund, again account8 in metamask
9. now click on balance shows the correct balance
and now when I click on withdraw I get rpc-error
even I click on withdraw the metamask shows the account8 it is connected to

I am very confused here, what account/address do I need to check fro the balance?
when foundry forge is run from the backend, I get that the owner who can withdraw is the address that created the account. 
But who is owner here in metamask that can withdraw.

Appreciate some help here.

**Comment by Niferu** [comment]
Hi @medhananda 

yea the problem here is understanding who is contract owner. If you are using foundry script it might be foundry contract instead of your wallet, who is contract owner. Just make sure that you are owner of the contract by changing your contract code or changing script code you are using and also add getOwner() function to your contract to see who the real owner is. You should also run your deploy script with -vvvv to see what address was deploying that contract which makes him owner. It is pretty annyoing on the beginning but later on you will get the understanding, so I highly recommend you printing all the data to console while you are running scripts/deploying etc. it is veeery helpful always. Good luck and let me know if you have fixed issue :)

**Reply by Niferu** [reply]
If you have any troubles with editing code let me know I can help you, cheers

**Comment by saintjer2224** [comment]
I'm getting the same error. Can fund successfully but withdraw always fails due to json-rpc error. @Nlferu , can you clarify a bit further on your advice above? Thanks 
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/108886381/5bf5345f-1714-4a79-82fd-7d2853035634)

**Reply by saintjer2224** [reply]
I followed the steps performed by Patrick and in the tutorial it shows he was able to withdraw successfully, not me though.

**Reply by Niferu** [reply]
Hi @saintjer2224 

Please make new discussion tab as this is already dead. Post your question, error and repo there and mark me, so I will be able to help you clarifying whats going on :) It will be helpful for others too if we solve it together

**Comment by billyrg131** [comment]
I have another solution to solve the internal JSON-RPC error. Go to the Metamask extension homepage. At the top-right, click on the Network icon to see enabled networks. Click on Manage Permissions, from there click on Edit in the Use Your Enabled Networks section. From there, select GoChain testnet(which in short is the Anvil local chain) and click on Update to enable only that network. Then you might be able to run fund() from the deployed FundMe on Anvil.
<img width="401" height="592" alt="Screenshot 2025-11-20 at 16 01 37" src="https://github.com/user-attachments/assets/88f267d8-e386-4508-b0a1-5584bb271611" />
<img width="401" height="592" alt="Screenshot 2025-11-20 at 16 01 56" src="https://github.com/user-attachments/assets/8992efd8-0d93-4b36-bfe6-3a2eadf995cd" />
<img width="401" height="592" alt="Screenshot 2025-11-20 at 16 02 05" src="https://github.com/user-attachments/assets/021a3151-7067-43f0-be2e-bfc74b1c4d7d" />
<img width="401" height="592" alt="Screenshot 2025-11-20 at 16 03 25" src="https://github.com/user-attachments/assets/6238b588-27a6-4629-8763-eb83a4272fca" />

**Reply by EngrPips** [reply]
Thanks for the heads-up
